src/utils/c_utils/src/string_c_utils.c

```diff
@@ -438,3 +438,43 @@ size_t ADUC_Safe_StrCopyN(char* dest, const char* src, size_t destByteLen, size_
 
     return numSrcCharsToCopy;
 }
+
+/**
+* @brief Allocate a new str that is a copy of the src string with given max number of char.
+* @param dest The addr of ptr that will point to newly allocated string copy.
+* @param src The source string.
+* @param srcLen The char length of source string (not including null terminator)
+* @return 0 on success, -2 on invalid argument, and -1 for other errors.
+*/
+int ADUC_AllocAndStrCopyN(char** dest, const char* src, size_t srcLen)
+{
+    int res = -1;
+
+    size_t num_copied = -1;
+    char* tmpDest = NULL;
+
+    if (dest == NULL || srcLen == 0)
```

Comment:
Add IsNullOrEmpty(src)

Reply:
won't fix. There is unit test for empty src and it's safe due to use of ADUC_Safe_StrCopyN below and it will fail if srcLen was > 0. Anyways, these functions are revamped a bit in a later stack PR for branch that was merged into user/jw-msft/v2-procupd